optuna/samplers/_tpe/sampler.py

```diff
@@ -477,59 +476,75 @@ def _sample(
             self._constraints_func is not None,
         )
 
-        below = self._get_internal_repr(below_trials, search_space)
-        above = self._get_internal_repr(above_trials, search_space)
+        mpe_below = self._build_parzen_estimator(
```

Comment:
Is there a need to divide the part that builds the Parzen estimator into a function? Simply increasing the depth of function calling will, in a sense, increase the complexity of the code. The function _build_parzen_estimator does not seem to obscure processes too complex to separate. Personally, it seems that the existence of the handle_below argument makes this method confusing.